cypress/support/fragments/data_import/dataImportViewAllPage.js

```diff
@@ -15,8 +26,239 @@ export default {
+      this.checkRowsCount(count);
+    });
   },
 
   options: ['Keyword (ID, File name)', 'ID', 'File name'],
+
+  errorsInImportStatuses: ['No', 'Yes'],
+
+  singleRecordImportsStatuses: ['No', 'Yes'],
+
+  resetAllFilters() {
+    cy.do(Button('Reset all').click());
+
+    // After resetting all filters, we need to sort MultiColumnList
+    // Otherwise, server cannot parse request params and returns error with 422 status
+    // In this case, sort by completed date in ascending order
+    cy.do(MultiColumnListHeader('Ended running').click());
+  },
+
+  filterJobsByErrors(filter) {
+    if (filter === 'Yes') {
+      return cy.do(CheckBox({ id: 'clickable-filter-statusAny-error' }).click());
+    } else {
+      return cy.do(CheckBox({ id: 'clickable-filter-statusAny-committed' }).click());
+    }
+  },
+
+  filterJobsByDate({ from, end }) {
+    cy.do([
+      Accordion({ id: 'completedDate' }).clickHeader(),
+      TextField({ label: 'From' }).fillIn(from),
+      TextField({ label: 'To' }).fillIn(end),
+      Button('Apply').click()
+    ]);
+  },
+
+  filterJobsByJobProfile(jobProfile) {
+    cy.do([
+      Accordion({ id: 'profileIdAny' }).clickHeader(),
+      Selection({ value: 'Choose job profile' }).open(),
+      SelectionList().select(jobProfile)
+    ]);
+  },
+
+  filterJobsByUser(user) {
+    cy.do([Accordion({ id: 'userId' }).clickHeader(),
+      Selection({ value: 'Choose user' }).open(),
+      SelectionList().select(user)
+    ]);
+  },
+
+  filterJobsByInventorySingleRecordImports(filter) {
+    if (filter === 'Yes') {
+      cy.do(CheckBox({ id: 'clickable-filter-singleRecordImports-yes' }).click());
+    } else {
+      cy.do(CheckBox({ id: 'clickable-filter-singleRecordImports-no' }).click());
+    }
+  },
+
+  waitUIToBeFiltered() {
+    // FIX: Find better approach for waiting MultiColumnList to be filtered
+    // when filtered, ui takes some time to be filtered
+    // intercepting and waiting API response is not reliable
+    // because, API may return response faster while ui is being filtered
+
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    return cy.wait(1800);
+  },
+
+  getMultiColumnListCellsValues(cell) {
+    const cells = [];
+
+    // get MultiColumnList rows and loop over
+    return cy.get('[data-row-index]').each($row => {
+      // from each row, choose specific cell
+      cy.get(`[class*="mclCell-"]:nth-child(${cell})`, { withinSubject: $row })
+        // extract its text content
+        .invoke('text')
+        .then(cellValue => {
+          cells.push(cellValue);
+        });
+    })
+      .then(() => cells);
+  },
+
+  checkByReverseChronologicalOrder() {
+    this.getMultiColumnListCellsValues(5).then(cells => {
+      // convert each cell value to Date object
+      const dates = cells.map(cell => new Date(cell));
+
+      // create new array from the dates and sort this array in descending order
+      const sortedDates = dates.slice(0).sort((a, b) => b - a);
```

Comment:
is it more convenient way instead of creation of new array through spread operator?

Reply:
@Javokhir12  have a look please